### ISSUE TITLE
Linkerd 2.11.1 controller pods are not running with linkerd-cni option in self-hosted k8s on AWS EC2 + Calico eBPF

### ISSUE THREAD

**Issue by brkane** [linkerd/linkerd2#7786]
### What is the issue?

Very similar to #7493, a fresh installation of linkerd 2.11.1 control plane is stuck in `ContainerCreating` and `CrashLoopBackOff`.

```
❯ kg pod -n linkerd
NAME                                      READY   STATUS              RESTARTS   AGE
linkerd-destination-6bf7f47c5-tbd8z       0/4     ContainerCreating   0          18m
linkerd-identity-cbd8f4795-65sq6          0/2     CrashLoopBackOff    8          18m
linkerd-proxy-injector-5c58568495-4bnhx   0/2     ContainerCreating   0          18m
```

### How can it be reproduced?

1. Set up a k8s cluster api version 1.21.9, kublet version 1.21.9, calico CNI version 3.21.3 in eBPF mode on AWS EC2
2. Install linkerd-cni version 2.11.1 `linkerd install-cni | kubectl apply -f -`
3. Install linked 2.11.1 `linkerd install --linkerd-cni-enabled | kubectl apply -f -`

### Logs, error output, etc

https://gist.github.com/brkane/d14ae9e1d221a78dea475fe156c1ec1c

### output of `linkerd check -o short`

```
Linkerd core checks
===================

linkerd-existence
-----------------
/ No running pods for "linkerd-destination"
```
The check hangs on this.

### Environment

- Kubernetes: api version 1.21.9, kublet version 1.21.9
- Cluster environment: AWS EC2
- OS: Ubuntu 20.04
- Linkerd version: 2.11.1
- Calico version: 3.21.3 in eBPF mode

### Possible solution

The issue appears to be because we are using calico in eBPF mode. When the control plane services reach out to the api server, they use the https://10.96.0.1:443 endpoint provided by KUBERNETES_SERVICE_HOST and KUBERNETES_SERVICE_PORT. However the backend api-services are actually listening on port 6443. When the TCP connection is opened, calico rewrites the connection from `10.96.0.1:443` to `<api pod ip>:6443`, as defined in the `kubernetes.default` service. This happens _before_ the PROXY_INIT_OUTPUT chain is evaluated, so the connection is directed to the linkerd-proxy sidecar, which is not fully up.

If I go onto the node and manually add in the iptables override rule for port 6443 `sudo ip netns exec cni-673c552a-99ed-7138-7242-6f3acb8695c5 iptables -t nat -I PROXY_INIT_OUTPUT 3 -p tcp -m multiport --dports 6443 -j RETURN`, the identity service is able to come up.

```
❯ kg service -n default kubernetes -o yaml
apiVersion: v1
kind: Service
metadata:
  creationTimestamp: "2019-09-03T19:33:38Z"
  labels:
    component: apiserver
    provider: kubernetes
  name: kubernetes
  namespace: default
  resourceVersion: "150"
  uid: ba8c8e8f-ce81-11e9-b971-02fc42ee36fa
spec:
  clusterIP: 10.96.0.1
  clusterIPs:
  - 10.96.0.1
  ipFamilies:
  - IPv4
  ipFamilyPolicy: SingleStack
  ports:
  - name: https
    port: 443
    protocol: TCP
    targetPort: 6443
  sessionAffinity: None
  type: ClusterIP
status:
  loadBalancer: {}
```

### Would you like to work on fixing this bug?

maybe

**Comment by ghouscht** [comment]
FYI: Looks like we face the same issue with Cilium (kube-proxy replacement enabled). Installed the linkerd-cni first and then linkerd with the `linkerd` CLI version 2.11.1.

**Comment by mibmo** [comment]
I am also having this issue with Cilium on a freshly-made Scaleway-managed cluster. No configuration of the `linkerd-cni` being enabled and running on stable or edge work. Also tried installing with the proxy-await & no dice.
```
/tmp # linkerd check -o short
Linkerd core checks
===================

linkerd-existence
-----------------
/ pod/linkerd-destination-6958c898bf-sswkx container sp-validator is not ready
```

**Comment by mateiidavid** [comment]
@mibmo if you're having trouble, I suggest filing a new issue with as many details as possible about your env (including which Linkerd version(s) you've tried using) and possible repro steps. Note that some CNIs re-write packet headers, depending on how they were configured; I'd check that's not the case for you.

**Comment by kandeshvari** [comment]
Workaround:

add port `6443` to `--outbound-ports-to-ignore` option in `linkerd-init` init container for all three linkerd deployments

**Comment by mateiidavid** [comment]
That would work, yes. I suspect for `linkerd-cni` you could do something similar by annotating the control plane components with `config.linkerd.io/skip-outbound-ports: "6443"`. That should have a similar effect.

**Comment by alpeb** [comment]
Fixed by #9841

The latest versions `edge-22.12.1` and `stable-2.12.3` have now port 6443 being skipped by the iptables rules to work around this issue. Please give them a try a re-open this issue it the problem persists.
